### ISSUE TITLE
iced_web: Add an option to select the element

### ISSUE THREAD

**Issue by MarcAntoine-Arnaud** [iced-rs/iced#491]
On running the main application, it can be awesome to be able to choose something else than the `body` [here](https://github.com/hecrj/iced/blob/master/web/src/lib.rs#L182).
Is it possible to add a parameter to the `run` with an `Option<&str>` to allow to select an element by it's ID where we can put the renderer.

It can be awesome to do that to allow integrated view inside other frameworks like React, AngularJS, Vue etc.

Marc-Antoine

**Comment by Kaiden42** [comment]
I would not like to consider adding parameters to the `run` function as this would cause a breaking change for everyone currently using iced (web). 🤔

Instead I would prefer to add this to the application settings instead. So for those who already use `Settings::default()` or similar nothing would change. 🙂

But there is one thing I don't quite understand here. `iced_web` has no settings, only flags. In the documentation of iced's `Settings` it is mentioned that, for example, the [`window`](https://github.com/hecrj/iced/blob/master/src/settings.rs#L9) field is ignored on the web, but as mentioned before there are no settings for web. 🤨

Maybe someone who knows a bit more about this project can help me here. Is there a reason why iced_web has no `Settings`?

**Comment by MarcAntoine-Arnaud** [comment]
Hi @Kaiden42,

Thank for sharing your point of view.
It sounds better to pass this king of parameter via [`Settings`](https://github.com/hecrj/iced/blob/master/src/settings.rs#L6) structure, I'm 100% agreed.

The comment on [`window`](https://github.com/hecrj/iced/blob/master/src/settings.rs#L9) is just there to said the [`window::Setting`](https://github.com/hecrj/iced/blob/master/src/window/settings.rs#L5) parameters are not used in the web case.

So to improve that I suggest to add a field in `window::Settings` - only for wasm32 target to add an `HtmlElement`.
So it can look like:
```rust
#[derive(Debug, Clone)]
pub struct Settings {
    /// The initial size of the window.
    pub size: (u32, u32),

    ....

    /// The icon of the window.
    pub icon: Option<Icon>,

    /// Optional HtmlElement to put the "window" into it.
    #[cfg(target_arch = "wasm32")]
    pub html_element: Option<HtmlElement>
}
```

Marc-Antoine

**Comment by hecrj** [comment]
An `Application` is meant to take control of the full document, as it can control the `title`.

The best way to satisfy this use case would be to create a different trait without a `title` method and an additional argument for `run`.

**Comment by Kaiden42** [comment]
Oh, yes, you're right. I'm sorry I completely forgot the title.

I would like to work on this issue. However, I have to come up with something clever how to share some of the logic between the two traits to avoid code duplication, as these are very similar.

I already have an idea and will work on a draft. :)

**Comment by hecrj** [comment]
@Kaiden42 Don't worry too much about code duplication. It's much easier to deal with duplicated code than with a leaky abstraction.

Most of the code in `Application::run` is boring code that connects mutliple parts together (i.e. plumbing). We want to keep this code flexible.

Basically, don't be afraid to copy-paste! We can come up with an abstraction once we are confident we are not dealing with coincidental duplication.

**Comment by Kaiden42** [comment]
Alright! I now have two drafts. One with code duplication and, as I wanted to, one without. However, I am unhappy with the second draft and I do not like it. Therefore I will stick with the first draft for now.

**Comment by hecrj** [comment]
Closed by #517.
